src/main/java/de/retest/recheck/NoGoldenMasterActionReplayResult.java

```diff
@@ -40,7 +40,7 @@ private static StateDifference toStateDifference( final SutState actual ) {
 	}
 
 	@Override
-	public Set<LeafDifference> getDifferences( final ShouldIgnore ignore ) {
+	public Set<LeafDifference> getDifferences( final Filter ignore ) {
```

Comment:
Without looking at the entire PR: You should also change variable/parameter/… names, like here from `ignore` to `filter`. Same e.g. in `CompundFilter` ctor, where it says:

```java
public CompoundFilter( final List<Filter> shouldIgnores ) {
    filter = shouldIgnores;
}
```

This should be `filter` as well. Please change this globally.